packages/client/hmi-client/src/components/workflow/tera-workflow-node.vue

```diff
@@ -21,20 +21,68 @@
 
 <script setup lang="ts">
 import { WorkflowNode } from '@/types/workflow';
-import { ref, onMounted } from 'vue';
+import { ref, computed, onMounted, onBeforeUnmount } from 'vue';
+
+const emit = defineEmits(['dragging']);
 
 const props = defineProps<{
 	node: WorkflowNode;
 }>();
 
-const nodeStyle = ref({
+const nodeStyle = computed(() => ({
 	minWidth: `${props.node.width}px`,
 	minHeight: `${props.node.height}px`,
 	top: `${props.node.y}px`,
 	left: `${props.node.x}px`
+}));
+
+const workflowNode = ref<HTMLElement>();
+
+let tempX = 0;
+let tempY = 0;
+let dragStart = false;
+
+const startDrag = (evt: MouseEvent) => {
+	console.log('start', evt.x, evt.y);
+	tempX = evt.x;
+	tempY = evt.y;
+	dragStart = true;
+};
+
+const drag = (evt: MouseEvent) => {
+	if (dragStart === false) return;
+
+	const dx = evt.x - tempX;
+	const dy = evt.y - tempY;
+
+	emit('dragging', { x: dx, y: dy });
+
+	tempX = evt.x;
+	tempY = evt.y;
+};
+
+const stopDrag = (evt: MouseEvent) => {
+	console.log('end', evt.x, evt.y);
+	tempX = 0;
+	tempY = 0;
+	dragStart = false;
+};
+
+onMounted(() => {
+	if (!workflowNode.value) return;
+
+	workflowNode.value.addEventListener('mousedown', startDrag);
+	document.addEventListener('mousemove', drag);
```

Comment:
Not too happy about hooking into document listener for every workflow-node, but it should work and we are not expected to have hundreds of nodes anyway.

Reply:
Hmm I am fiddling with `{defineExpose}` if you want to go that route.

Reply:
@shawnyama I am not sure what you are trying to say - can you elaborate or have a small code-snippet that we can use to base the conversation?

Reply:
Hmm I see, I have not played with defineExpose before so I am not sure how to work with it or the caveats. Might also able to do this as a currying function going parent=>child instead of child=>parent

Anyway I am going to merge to PR and we can have this as a side discussion.